File: api/Models/Supplier.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace api.Models
{
    [Index(nameof(Ruc), IsUnique = true)]
    public class Supplier
    {
        public int Id { get; set; }
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = default!;
        [Required]
        [MaxLength(20)]
        public string Ruc { get; set; } = default!;
        [Required]
        [MaxLength(15)]
        public string Phone { get; set; } = default!;
        [Required]
        [MaxLength(50)]
        public string Email { get; set; } = default!;
        [Required]
        [MaxLength(150)]
        public string Address { get; set; } = default!;
        public ICollection<BuyOrder> BuyOrders { get; set; } = new List<BuyOrder>();
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
    }
}

File: api/app/models/supplier.ts
import { DateTime } from 'luxon';
import { BaseModel, column, hasMany } from '@adonisjs/lucid/orm';
import BuyOrder from './buy_order.js';
import type { HasMany } from '@adonisjs/lucid/types/relations';

export default class Supplier extends BaseModel {
  @column({ isPrimary: true })
  declare id: number;

  @column()
  declare name: string;

  @column()
  declare ruc: string;

  @column()
  declare phone: string;

  @column()
  declare email: string;

  @column()
  declare address: string;

  @column.dateTime({ autoCreate: true })
  declare createdAt: DateTime;

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  declare updatedAt: DateTime;

  @column.dateTime()
  declare deletedAt: DateTime | null;

  @hasMany(() => BuyOrder)
  declare buyOrders: HasMany<typeof BuyOrder>;
}
